README.md
## How-To's  

Repository with technical indications to install, configure any interesting stuff. 

### Resources

1. [Install Ardour in Ubuntu 20.04](doc/ardour_in_ubuntu_20.04.md)
2. [NewPassword Generator Java test](src/NewPasswordGenerator.java)
3. [Python guides and samples](doc/python_docs_samples.md)
4. [Customizing/styling a Windows and Linux Prompt](doc/styled_win_linux_prompt.md) 
5. [Install patched fonts (with glyphs) required for styled Win/Linux Prompts and modern Terminals](doc/patched_fonts.md)
6. [Working with TMux](doc/working_with_tmux.md)
7. Git guides
  * [Github - Frequent commands](doc/git_frequent_commands.md)
  * [Github - Authentication, Tokens and Credential Manager](doc/git_auth_guide.md)
  * [GitHub's Hub wrapper - Install and configure](src/git_and_hub_setting_in_linux.sh)
  * [Create new Git repo from sub-folder of existing repo preserving history](doc/git_subfolder_to_repository.md)
  * Install Git and Bash emulator in Windows:  
    - [Git for Windows](https://gitforwindows.org/)
    - Download `Git-<version>-64-bit.exe` from [https://github.com/git-for-windows/git/releases/](https://github.com/git-for-windows/git/releases/) or [https://git-scm.com/download/win](https://git-scm.com/download/win), execute it and follow all instructions.
  * Install GitHub Hub for Windows:  
    - [GitHub Hub](https://hub.github.com/) is a extension to Git CLI that helps to manage GitHub from your Terminal.
    - Download `hub-windows-amd64-<version>.zip` from [https://github.com/github/hub/releases](https://github.com/github/hub/releases), unzip it and install it in your system.
  * [Signing Git commits and tags with GPG and SSH keys](doc/git_signing.md) 
8. [Getting CPU and GPU temperature in Ubuntu 19.04](doc/getting_temperature_cpu_gpu_hd_in_ubuntu.md)
9. [Install Logitech Unifying (Keyboard and mice) in Ubuntu 19.04](doc/install_logi_unifying_in_ubuntu_19_04.md)
10. Screen Tools:
  - [Install Asus MB168b portable DisplayLink compatible screen in Ubuntu 18.04](doc/screen_mb168b_install_in_ubuntu.md)
  - [Take screenshots silently in Ubuntu with xfce4-screenshooter](doc/screen_shooter_silent.md)
  - [Change the brightness level of a screen](doc/screen_change_brightness_level.md)
  - [Screenshot and annotation Tools](doc/screenshot_annotation_tools.md) 
11. [Working with Tmux](doc/working_with_tmux.md)
12. [File sharing through Samba(SMB)](doc/install_and_config_samba.md)
13. Cloud and Infrastructure as Code (IaC) examples:
  - [Terraforms examples](doc/iac_terraform_examples.md)
  - [CloudFormation examples](doc/iac_cloudformation_examples.md)
14. [Install TightVNC - VNC remote access on Ubuntu 20.04](doc/install_tightvnc.md)
15. [Install Apache Guacamole - Remote access Gateway on Ubuntu 20.04](doc/install_apache_guacamole.md)
16. [Create booteable USB Ubuntu](doc/booteable_usb_on_ubuntu.md)  
17. [Google Drive on Ubuntu with XFCE](doc/google_drive_on_linux.md)  
18. [Regex examples in VSCode](doc/regex_examples.md)
19. [Generate and import SSH Pub Key to all AWS Regions](doc/import_ssh_keys_to_aws_regions.md)
20. [Disabling sleeping when close laptop lid](doc/disable_sleeping_when_close_laptop_lid.md)
21. Docker & Docker Compose
  - [Docker install and commands](doc/docker_useful_commands.md)
  - [Docker Compose install and example](doc/docker-compose-install/)
22. [Raspberry Pi - Guides]
  - [Install Raspbian OS and Ubuntu 64bits on Raspberry Pi 3B+ in headless mode](doc/raspberry_pi_getting_started.md)
  - [Install Code-Server on RPi](doc/install_code_server_on_headless_rpi.md)
  - [Install Pi-Hole on RPi](doc/pi_hole_guide.md)
23. [Install AutoFirma Java App in Ubuntu 19.10](doc/install_autofirma_app_in_ubuntu19.10.md)
24. [Installation of WSO2 MB, Apache Qpid, RabbitMQ and Apache ActiveMQ on CentOS](doc/install_wso2mb_qpid_rabbitmq_activemq.md)
25. [Issue Certificates with MKCert](doc/issue_certs_with_mkcert.md)
26. GitHub Pages - Building static website with Jekyll and Hugo:
  - [Create static web with Jekyll](doc/github_pages_jekyll_create_a_static_web.md)
  - [Create static web with Hugo](doc/github_pages_hugo_create_a_static_web.md)
  - [Publish new content with Hugo](doc/github_pages_hugo_publish_content.md)
  - [Publish new content on Waskhar Project](doc/github_pages_hugo_publish_content_waskhar.md)
27. [Installing and using NMAP](doc/nmap_commands.md)
28. [Installing and uninstalling IDE and DevOps Tools for multiple platforms](doc/ide_and_devops_tools.md)
29. [CI/CD GitHub workflows for NodeJS and TypeScript applications](doc/github-workflow-for-nodejs-typescript-apps.md)
30. [Installing IPFS Cluster CLI](doc/ipfs_cluster_ctl_commands.md)
31. [RESTful API - Generating Documentation and OpenAPI specs from Golang API](src/go-rest-openapi-sample/)
32. [RESTful API - Security checking using Spectral Linter](doc/api_security_checks.md)
33. [Bitnami Sealed Secrets on Google Kubernetes Engine](doc/bitnami_sealed_secrets_on_gke_guide.md)
34. [Setting iMac (24" early 2009) fan control for Ubuntu](doc/imac_early_2009_fan_control_on_ubuntu.md)
34. [Turn off Macbook/iMac startup sound on Ubuntu](doc/turn_off_mac_startup_sound_on_ubuntu.md)


CHANGELOG.md
How-To's changelog

# [1.0.0](https://github.com/chilcano/how-tos/compare/v0.2.0...v1.0.0) (2023-04-11)


### Features

* add gh action to test sharing vars between jobs ([62ffb2b](https://github.com/chilcano/how-tos/commit/62ffb2bb659e18da1a44aca65b4d361b8a815166))
* add software release bpmn process ([a77ceb7](https://github.com/chilcano/how-tos/commit/a77ceb7f3231fea825f0cd3158cf8d3b27375e68))
* **ci:** update workflow to use changelog plugin ([7fa4907](https://github.com/chilcano/how-tos/commit/7fa4907f6e083adbacf8adfe0e8186bf4a81f3ed))
* **ci:** update workflow to use choice input param ([d0a17ae](https://github.com/chilcano/how-tos/commit/d0a17ae32df19459093213425b14f55e52d25f21))
* **ci:** update workflow to use input params ([022647c](https://github.com/chilcano/how-tos/commit/022647cc7d6735d0d63f6be11bff41de5430152d))
* **processes:** add deploy new service diagram ([3e9cd29](https://github.com/chilcano/how-tos/commit/3e9cd29f2b99e8a5dba1c5fe3ec6878c7e233c76))
* **processes:** add diagram to request new software ([21b039c](https://github.com/chilcano/how-tos/commit/21b039c6fed06049d00db42223dca50b7fdf1e8b))
* **processes:** add draft diagram to deploy new service ([9f5accc](https://github.com/chilcano/how-tos/commit/9f5accc9d13c8e2c8356d1a18724370d057b01e3))
* **processes:** update offboarding diagram ([9a1b40e](https://github.com/chilcano/how-tos/commit/9a1b40eed03346ed16d874c0a30bfaf58d2c85e8))
* **sealed-secrets:** guide updated ([f9339d4](https://github.com/chilcano/how-tos/commit/f9339d4d36cb3f68e04f85240fe2f8d82ebaaf6d))


### BREAKING CHANGES

* **processes:** The deploy new service is still in draft. This process
is not recommended for production.
* **processes:** The Deploy new service process is incomplete and it should not be used at the moment.


src/code_server_tls.sh
#!/bin/bash

echo "+--------------------------------------------------------+"
echo "¦      Exposing VSCode Server over TLS with Caddy        ¦"
echo "+--------------------------------------------------------+"

NOW1=$(date +"%y%m%d.%H%M%S")


## ------------------ install caddy

#echo "deb [trusted=yes] https://apt.fury.io/caddy/ /" | sudo tee -a /etc/apt/sources.list.d/caddy-fury.list
#sudo apt update
#sudo apt install -y caddy
# ...
#Unpacking caddy (2.1.1) ...
#Setting up caddy (2.1.1) ...
#Created symlink /etc/systemd/system/multi-user.target.wants/caddy.service → /lib/systemd/system/caddy.service.
#System has not been booted with systemd as init system (PID 1). Can't operate.
#Failed to connect to bus: Host is down
#System has not been booted with systemd as init system (PID 1). Can't operate.
#Failed to connect to bus: Host is down


## --------------- backup Caddyfile
#sudo chown -R root:root Caddyfile
#sudo mv /etc/caddy/Caddyfile "/etc/caddy/Caddyfile.${NOW1}" 
## copy Caddyfile
#sudo mv Caddyfile /etc/caddy/Caddyfile

# ---------------- this cmd will not work in WSL
#sudo systemctl reload caddy

### -------------- will work in WSL
#caddy reverse-proxy --from :8443 --to 127.0.0.1:8080

#2020/07/13 18:30:06.124 WARN    admin   admin endpoint disabled
#2020/07/13 20:30:06 [INFO][cache:0xc0002012c0] Started certificate maintenance routine
#2020/07/13 18:30:06.130 INFO    tls     cleaned up storage units
#2020/07/13 18:30:06.131 INFO    autosaved config        {"file": "/home/rogerc/.config/caddy/autosave.json"}
#Caddy proxying http://:8443 -> http://127.0.0.1:8080



## ------------------ load caddy.json from cmd, also works in WSL
sudo caddy start
curl localhost:2019/load -X POST -H "Content-Type: application/json" -d @/home/rogerc/gitrepos/how-tos/src/caddy.json
#curl localhost:2019/load -X POST -H "Content-Type: application/json" -d @/home/pi/gitrepos/how-tos/src/caddy.json
caddy reverse-proxy --from :8443 --to 127.0.0.1:8080

# ------------------ query the config loaded
# curl localhost:2019/config/ | jq -C

printf ">> VSCode Server is running over TLS. \n"


## Certs created by Caddy

#$ sudo ls -la /var/lib/caddy/.local/share/caddy/pki/authorities/local
#total 24
#drwx------ 2 caddy caddy 4096 Jun 23 18:31 .
#drwx------ 3 caddy caddy 4096 Jun 23 18:31 ..
#-rw------- 1 caddy caddy  680 Jul  1 10:42 intermediate.crt
#-rw------- 1 caddy caddy  227 Jul  1 10:42 intermediate.key
#-rw------- 1 caddy caddy  627 Jun 23 18:31 root.crt
#-rw------- 1 caddy caddy  227 Jun 23 18:31 root.key